nir/src/main/scala/scala/scalanative/nir/serialization/BinarySerializer.scala

```diff
@@ -3,18 +3,31 @@ package nir
 package serialization
 
 import java.net.URI
-import java.nio.ByteBuffer
+import java.io.{DataOutputStream, OutputStream}
 import java.nio.charset.StandardCharsets
 import scala.collection.mutable
 import scala.scalanative.nir.serialization.{Tags => T}
 
-final class BinarySerializer(buffer: ByteBuffer) {
-  import buffer._
+final class BinarySerializer {
+  private[this] val bufferUnderyling = new JumpBackByteArrayOutputStream
+  private[this] val buffer           = new DataOutputStream(bufferUnderyling)
 
   private[this] var lastPosition: Position = Position.NoPosition
   private[this] val fileIndexMap           = mutable.ListMap.empty[URI, Int]
 
-  final def serialize(defns: Seq[Defn]): Unit = {
+  // Methods were renamed in order to not pollute git blame history.
+  // Original implementation used ByteBuffers
+  import buffer.{
+    write => put,
+    writeDouble => putDouble,
+    writeFloat => putFloat,
+    writeInt => putInt,
+    writeChar => putChar,
+    writeLong => putLong,
+    writeShort => putShort
+  }
+  import bufferUnderyling.currentPosition
+  def serialize(defns: Seq[Defn], outputStream: OutputStream): Unit = {
```

Comment:
We deserve a blank line, here.
```suggestion
  import bufferUnderyling.currentPosition
  
  def serialize(defns: Seq[Defn], outputStream: OutputStream): Unit = {
```